### PR TITLE
Evolution use local servers

### PR DESCRIPTION
**Pull request by czerw**
Fix poo#19320: Local imap/pop3, smtp servers were setup for evolution
tests. The dependency on the company infrastructure located in APAC
region was removed.

Several evolution tests (evolution_mail_imap, evolution_mail_pop, evolution_timezone_setup, evolution_meeting_imap and evolution_meeting_pop) fail from time time, because of usage of remote infrastructure.

Dovecot and postfix servers were setup as local backends
Dovecot is not part of the SLED, SLE repository was added.

Verification:
SLED12-SP3: http://10.100.12.105/tests/1016#
SLED12-SP2: http://10.100.12.105/tests/1013#

It should cover these progress issues:
19320,18772,19054,15134,17596,18266

### CODE REVIEW

File: tests/x11regressions/evolution/evolution_prepare_servers.pm [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+use base "x11regressiontest";
+use testapi;
+use utils;
+
+sub run() {
+    select_console('root-console');
+    pkcon_quit;
+
+    # add SLES repository and install dovecot
+    zypper_call("ar http://download.suse.de/ibs/SUSE:/SLE-12:/Update/standard/ sle_server_repo");
```
Comment (okurz) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: I don't think it's a good idea to add this repo here. dovecot should be available from whatever repositories the system has already. What exactly do you want to achieve here?

Comment (czerw) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: Unfortunately, dovecot is not in SLED, it is only in SLES repositories. Do you have better idea how to get it to SLED tests?

Comment (okurz) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: I see, you want dovecot for local testing. I am not sure how stable this will be in the long run. Hm, I don't have a better idea. I guess this needs adaption for SLE15 at least later on.

**Comment by okurz** [comment]
+1

File: tests/x11regressions/evolution/evolution_prepare_servers.pm (hunk 1 above)
Comment (czerw) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: I was thinking about SLE15 when i was adding that line, but SLE15 is not out yet and I believe that adaptation for it will be easy. It works for 12SP3  (was tested, it use same base as 12SP2 ) , which is going to be live soon. 
The other way could be to use multimachine tests, but it looks like overkill just for this purpose.

Comment (okurz) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: Agreed. 12SP3 *is* live, see the validation tests of the latest build: https://openqa.suse.de/tests/1057895

Comment (coolo) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: at least add a die 'not supported' if version_at_least('15') - or we will get very strange failures

Comment (dzedro) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: Can't be `SLE-12:` a variable that will be made from VERSION ?

Comment (czerw) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: It could be, but right we don't know, how it will be for SLE-15. So, I think that it will be better to make it dumb/obvious what to change afterwards.

Comment (dzedro) [inline tests/x11regressions/evolution/evolution_prepare_servers.pm]: It will be ultimate evolution test running on everything :feelsgood:

**Comment by coolo** [comment]
https://openqa.suse.de/tests/1065521#step/evolution_mail_imap/13 - the evolution test is not more reliable now :(

**Comment by czerw** [comment]
I'm aware about it, that is something which didn't occur during  testing on local openqa instance and it looks that it can be reproduced when is machine with worker more loaded.

**Comment by czerw** [comment]
> the evolution test is not more reliable now :(
No. Although it is edge case, it is evolution bug which happens randomly. When you enter address with localhost, automatic detection sometimes fail. I reproduced it on SLED12 and my Tumbleweed and my colleague did a test for me and confirmed it.

It can sound as non senses to enter address "user@localhost", but skip button should work. I will create a product bug today for it

**Comment by czerw** [comment]
@coolo  Bug was created https://bugzilla.suse.com/show_bug.cgi?id=1049387 .  Should I update the test with workaround and soft fail warning?

**Comment by okurz** [comment]
IMHO always when we can we should update with a workaround and bugref. This will also help with reviewing as it is ensured the correct bug reference is stated when we already know the bug.
